frontend/tests/features/collection.test.ts

```diff
@@ -33,9 +34,13 @@ describe("Collection", () => {
       await goToPage(
         TEST_URL + ROUTES.PRIVATE_COLLECTION.replace(":id", collectionId)
       );
-      await expect(page).not.toHaveSelector(getText(TEST_COLLECTION.name), {
-        timeout: TIMEOUT_MS,
-      });
+
+      // wait five seconds for cache to clear
+      await setTimeout(async () => {
```

Comment:
I remember in Jest, functions in setTimeout() don't work, so tests will always pass!

https://stackoverflow.com/questions/46077176/jest-settimeout-not-pausing-test/46077769

Also, using a hard coded wait time is typically a code smell, because it masks the underlying flakiness and at times makes a test run longer than needed

What if we retry the test until the condition is met or timeout?

E.g.,

```ts

// This replaces 5 second wait, and will retry every 200ms until it either passes, or 
// reaches max retries and throws
await retryAssert(async () => { 
  await expect(page).not.toHaveSelector(
    getText(TEST_COLLECTION.name + timestamp)
  )
}, { maxRetry: 25 })

export async function retryAssert(assert, { maxRetry = 10, waitForMs = 200 }) {
  let retry = 0;

  while (retry < maxRetry) {
    try {
      await assert();

      break;
    } catch (error) {
      retry += 1;

      await page.waitFor(waitForMs);
    }
  }

  if (retry === MAX_RETRY) {
    throw Error("retryAssert() assertion failed!");
  }
}
```